packages/framework-provider-aws/src/library/searcher-adapter.ts

```diff
@@ -32,62 +31,150 @@ export async function searchReadModel(
   return result.Items ?? []
 }
 
-function buildFilterExpression(filters: Record<string, Filter<any>>): string {
+function buildExpression(filters: FilterFor<any>): any {
+  const expresion = buildFilterExpression(filters)
+  const names = buildExpressionAttributeNames(filters)
+  resetAttrCount()
+  const values = buildExpressionAttributeValues(filters)
+  resetAttrCount()
+  return {
+    FilterExpression: expresion,
+    ExpressionAttributeNames: names,
+    ExpressionAttributeValues: values,
+  }
+}
+
+function buildFilterExpression(filters: FilterFor<any>): string {
```

Comment:
**Related to the proposal to delete the global `attrCount` array:**

```suggestion
function buildFilterExpression(filters: FilterFor<any>, usedPlaceholders: Array<string>): string {
```

Also, this parameter should be passed to the different recursive calls we have here